### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by maciej-zieba**

### CODE REVIEW

File: programs/exchange/src/utils.rs [hunk 1]
```diff
@@ -1605,4 +1605,320 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_set_mint_amount_safely(){
+        
+        // decrease is safety
```
Comment (matzayonc) [inline programs/exchange/src/utils.rs:1612]: is safe?

File: programs/exchange/src/utils.rs [hunk 2]
```diff
@@ -1605,4 +1605,320 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_set_mint_amount_safely(){
```
Comment (matzayonc) [inline programs/exchange/src/utils.rs:1610]: missing case of increase without error

File: programs/exchange/src/utils.rs [hunk 3]
```diff
@@ -1605,4 +1605,320 @@ mod tests {
...
+
+            let result = vault.set_mint_amount_safely(new_mint_amount);
+
+            assert!(result.is_err());
+            
+        }
+    }
+
+    #[test]
+    fn test_append_exchange_account(){
```
Comment (matzayonc) [inline programs/exchange/src/utils.rs:1643]: maybe case with second append to same account?

File: programs/exchange/src/utils.rs [hunk 4]
```diff
@@ -1605,4 +1605,320 @@ mod tests {
...
+                entry,
+            );
+
+            assert_eq!(exchange_account.head, 1);
+            assert_eq!(exchange_account.collaterals[(exchange_account.head-1) as usize], entry);
+        }
+    }
+
+    #[test]
+    fn test_remove_exchange_account(){
```
Comment (matzayonc) [inline programs/exchange/src/utils.rs:1672]: case with removing not last entry?

File: programs/exchange/src/utils.rs [hunk 5]
```diff
@@ -1605,4 +1605,392 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_set_mint_amount_safely(){
```
Comment (matzayonc) [inline programs/exchange/src/utils.rs:1610]: maybe a case of increase without error?

**Review by matzayonc — APPROVED** [review]

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**